test/test_nn.py

```diff
@@ -7737,6 +7739,22 @@ def test_smoothl1loss_negative_beta_not_supported(self):
         with self.assertRaises(RuntimeError):
             F.smooth_l1_loss(torch.randn(2, 2), torch.randn(2, 2), beta=-1.0)
 
+    def test_huber_loss_invalid_beta(self):
+        def _test_huber_loss_beta_error_helper(beta=1.0):
+            input, target = torch.randn(2, 2), torch.randn(2, 2)
+            loss = torch.nn.HuberLoss(beta=beta)
+            with self.assertRaises(RuntimeError):
+                loss(input, target)
+
+        def test_huber_loss_negative_beta():
+            _test_huber_loss_beta_error_helper(beta=-0.5)
+
+        def test_huber_loss_zero_beta():
+            _test_huber_loss_beta_error_helper(beta=0.0)
+
+        test_huber_loss_negative_beta()
```

Comment:
The extra indirection is done so that if a test fails, the specific failure value can be unambiguously identified in the stack trace. There would be a lot less duplication in general if we iterated through values, but then the stack trace wouldn't indicate exactly which value was the problem (it would just point to a line in the for loop).

The idea behind having these sub-tests here is that the different cases form one "conceptual test". This seems to be a fairly common pattern in `test_nn`, so I copied it here.

Reply:
Ok, so my comment here is most like: this is indented one time too many.
These are not actually "sub-tests" right?

> the specific failure value can be unambiguously identified in the stack trace. 

I am not sure how the 2 indirection help, wouldn't 1 indirection be enough to get that?